A5-grpclb-in-dns.md

```diff
@@ -0,0 +1,173 @@
+balancers are unreachable.  However, that fallback functionality is not
+yet implemented, so it will be the subject of future work.
+
+However, it is still possible for service owners to set up their DNS
+zone files to publish both types of addresses, so that they are prepared
+for an eventual future when this fallback functionality has been
+implemented.  For example:
+
+```
+$ORIGIN example.com.
+server              IN A 10.0.0.11
+                    IN A 10.0.0.12
+; grpclb for server.example.com goes to lb.example.com:1234
+_grpclb._tcp.server IN SRV 0 0 1234 lb
+; lb.example.com has 3 IP addresses
+lb                  IN A 10.0.0.1
+                    IN A 10.0.0.2
+                    IN A 10.0.0.3
+```
+
+For now, if both address records and SRV records are present for the
+same server name, the gRPC client will ignore the address records and
+return only the SRV records -- in other words, the result for a lookup
+of `server.example.com` will be exactly the same in this case as it
+would be in the previous example.
+
+In the future, when we do implement support for this kind of fallback,
+this lookup will result in the following addresses:
+
+```
+address=10.0.0.11:443, is_balancer=false, balancer_name=<unset>
+address=10.0.0.12:443, is_balancer=false, balancer_name=<unset>
+address=10.0.0.1:1234, is_balancer=true, balancer_name=lb.example.com
+address=10.0.0.2:1234, is_balancer=true, balancer_name=lb.example.com
+address=10.0.0.3:1234, is_balancer=true, balancer_name=lb.example.com
+```
+
+Note that port 443 is the default if not specified by the server name
+passed to the client library by the application.  If the server name
+passed in by the application does specify a port, that would be used for
+the resulting server addresses, but it would not affect the port used
+for the balancer addresses.
+
+## Rationale
+
+We considered one alternative, which was adding additional fields to a
+TXT record instead of using an SRV record.  However, the SRV record
+seems to be a much better fit for this purpose.
+
+## Implementation
+
+### C-core
+
+In C-core, this implementation will rely on use of the c-ares DNS
+library, which is being added in
+[grpc/grpc#7771](https://github.com/grpc/grpc/pull/7771).
+
+Specifically, we will change the c-ares resolver implementation to
+automatically look up the grpclb SRV records for the specified name.  If
+found, it will then look up the name(s) that the SRV records point to
+and return those as balancer addresses.
+
+Note that, due to platform support issues, we will initially *not*
+support the c-ares resolver under Windows or for Node.  Alternatives
+will need to be found for these environments.
+
+### Java
+
+Java will depend on JNDI, Netty DNS, dnsjava, or a another DNS library
+to do SRV record resolution. The existing name resolver, `DnsNameResolver`,
+will be modified to resolve the additional records and include them in
+the Attributes presented to the load balancer.
+
+### Go
```

Comment:
Thanks for the info.  I've updated the doc.

@a11r, I believe this is ready to be merged now.